src/init/cli/CliExtractor.ts

```diff
@@ -0,0 +1,14 @@
+import { AsyncHandler } from '../../util/handlers/AsyncHandler';
+
+/**
+ * Converts the input CLI arguments into an easily parseable key/value object.
+ *
+ * There are certain CLI parameters that are required before this class can be instantiated.
+ * These can be ignored by this class, but that does mean that this class should not error if they are present.
```

Comment:
The specific case for these requirements is if you use the strict mode of Yargs. The `CliExtractor` does not define the `-c/--config` parameter for example (since that is handled beforehand), but that parameter will be present in the array. Yargs in strict mode would error about this unknown parameter. Another solution I considered was removing those parameters from the array, but then that could be an issue if the `CliExtractor` was interested in them. And it is a bit iffy to edit the argv array.

Reply:
Okay; up to you.